src/test/java/teammates/common/datatransfer/questions/FeedbackMcqQuestionDetailsTest.java

```diff
@@ -225,4 +226,95 @@ public void testValidateResponsesDetails_noValidationError_errorListShouldBeEmpt
+        FeedbackMcqQuestionDetails mcqDetails = new FeedbackMcqQuestionDetails();
+        mcqDetails.setMcqWeights(Arrays.asList(1.22, -1.55));
+        mcqDetails.setMcqChoices(Arrays.asList("choice 1", "choice 2"));
+        mcqDetails.setHasAssignedWeights(false);
+        List<String> errors = mcqDetails.validateQuestionDetails();
+        assertEquals(FeedbackParticipantType.NONE, mcqDetails.getGenerateOptionsFor());
+        assertEquals(1, errors.size());
+        assertEquals(FeedbackMcqQuestionDetails.MCQ_ERROR_INVALID_WEIGHT, errors.get(0));
+    }
+
+    @Test
+    public void testValidateQuestionDetails_weightsNotEnabledButOtherWeightNotZero_errorReturned() {
+        FeedbackMcqQuestionDetails mcqDetails = new FeedbackMcqQuestionDetails();
+        mcqDetails.setMcqOtherWeight(0.5);
+        mcqDetails.setHasAssignedWeights(false);
+
+        List<String> errors = mcqDetails.validateQuestionDetails();
+        assertEquals(FeedbackParticipantType.NONE, mcqDetails.getGenerateOptionsFor());
+        assertEquals(2, errors.size());
+        assertEquals(FeedbackMcqQuestionDetails.MCQ_ERROR_NOT_ENOUGH_CHOICES
+                + FeedbackMcqQuestionDetails.MCQ_MIN_NUM_OF_CHOICES + ".", errors.get(0));
+        assertEquals(FeedbackMcqQuestionDetails.MCQ_ERROR_INVALID_WEIGHT, errors.get(1));
+    }
+
+    @Test
+    public void testValidateQuestionDetails_hasAssignedWeightsOtherEnabledNonZeroWeight_errorReturned() {
+        FeedbackMcqQuestionDetails mcqDetails = new FeedbackMcqQuestionDetails();
+        mcqDetails.setHasAssignedWeights(true);
+        mcqDetails.setOtherEnabled(true);
+        mcqDetails.setMcqOtherWeight(1.5);
+        mcqDetails.setMcqChoices(Arrays.asList("choice 1", "choice 2"));
+
+        List<String> errors = mcqDetails.validateQuestionDetails();
+        assertEquals(1, errors.size());
+        assertEquals(FeedbackMcqQuestionDetails.MCQ_ERROR_INVALID_WEIGHT, errors.get(0));
+    }
+
+    @Test
+    public void testValidateQuestionDetails_hasAssignedWeightsOtherEnabledNegativeWeight_errorReturned() {
+        FeedbackMcqQuestionDetails mcqDetails = new FeedbackMcqQuestionDetails();
+        mcqDetails.setHasAssignedWeights(true);
+        mcqDetails.setOtherEnabled(true);
+        mcqDetails.setMcqOtherWeight(-1.5);
+
+        List<String> errors = mcqDetails.validateQuestionDetails();
+        assertEquals(2, errors.size());
+        assertEquals(FeedbackMcqQuestionDetails.MCQ_ERROR_NOT_ENOUGH_CHOICES
+                + FeedbackMcqQuestionDetails.MCQ_MIN_NUM_OF_CHOICES + ".", errors.get(0));
+        assertEquals(FeedbackMcqQuestionDetails.MCQ_ERROR_INVALID_WEIGHT, errors.get(1));
+    }
+
+    @Test
+    public void testValidateQuestionDetails_hasAssignedWeightsNonEmptyWeights_errorReturned() {
+        FeedbackMcqQuestionDetails mcqDetails = new FeedbackMcqQuestionDetails();
+        mcqDetails.setHasAssignedWeights(true);
+        mcqDetails.setMcqWeights(Collections.singletonList(1.5));
```

Comment:
could i ask why a singletonList is used here instead of a normal list?

Reply:
I wanted to create an immutable list to have a fixed set of weights associated with the question. Also, using `singletonList` is slightly more readable considering there's only a single weight to be assigned.  Open to converting it into a normal list if needed :)

Reply:
ah i see! if i'm not wrong, the list of mcqWeights does not necessarily have to be immutable for this class, and furthermore I think it's better if we keep it consistent with the rest of the test cases, where Arrays.asList is used when setting the mcqWeights, hence I think we should convert it to a normal list